### PR TITLE
Implemented Edit User Profile Feature

### LINKED ISSUE #93: Need for an edit option in profile screen which launches a edit profile screen
Edit profile screen would allow the user to edit their username, display name, email and profile Image 

This screen might also have a delete account option for now would navigate it to a settings screen if generated in future

Hey @chandansgowda would like to work on it

### PR DESCRIPTION
**Pull request by vrundraval24**
Fixes #93 

**Features:**
- Change Profile Picture - With the options of 
- Select From Camera, 
- Gallery or 
- Remove Profile Picture.
- Change User Display Name
- Change Username
- Save changes
- One dialogue box to alert the user if there are Unsaved Changes

I have also created Logic for **deleting the old username** from the database as the user **creates a new one**. So the old username can be **re-used**.

Display name and Username are properly **validated** and **username availability** is also checked.

Here's a zip file containing all the screenshots and screen recordings.
[Edit Profile - Vrund Raval.zip](https://github.com/AOSSIE-Org/Resonate/files/12873456/Edit.Profile.-.Vrund.Raval.zip)

### CODE REVIEW

**Comment by vrundraval24** [comment]
Hello @chandansgowda, Please review it.

File: lib/controllers/edit_profile_controller.dart [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  TextEditingController imageController = TextEditingController();
+  TextEditingController nameController = TextEditingController();
+  TextEditingController usernameController = TextEditingController();
+
+  final GlobalKey<FormState> editProfileFormKey = GlobalKey<FormState>();
+
+  @override
+  void onInit() {
+    super.onInit();
+    storage = Storage(authStateController.client);
```
Comment (chandansgowda) [inline lib/controllers/edit_profile_controller.dart:39]: Get it from AppwriteService class.

File: lib/controllers/edit_profile_controller.dart [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  }
+
+  bool isDisplayNameChanged() {
+    if (nameController.text == oldDisplayName) {
+      return false;
+    }
+    return true;
+  }
+
+  bool isPfpChanged() {
```
Comment (chandansgowda) [inline lib/controllers/edit_profile_controller.dart:144]: Use full names

**Review by chandansgowda — CHANGES_REQUESTED** [review]
Small changes left. I've added comments.

**Comment by vrundraval24** [comment]
@chandansgowda I have completed all the requested changes. Please review it.

I am also attaching screenshots of the changes

Screenshot 2023-10-12 222535
Screenshot 2023-10-12 222527
Screenshot 2023-10-12 222549
Screenshot 2023-10-12 222603

**Pull request merged**